tools/kafkatool/offsets.go

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/alecthomas/kingpin/v2"
+	"github.com/twmb/franz-go/pkg/kadm"
+	"github.com/twmb/franz-go/pkg/kgo"
+)
+
+// OffsetsCommand allows inspecting offsets for a given topic.
+type OffsetsCommand struct {
+	getKafkaClient func() *kgo.Client
+	printer        Printer
+
+	topic      string
+	offsetType string
+	millis     int64
+}
+
+// Register is used to register the command to a parent command.
+func (c *OffsetsCommand) Register(app *kingpin.Application, getKafkaClient func() *kgo.Client, printer Printer) {
+	c.getKafkaClient = getKafkaClient
+	c.printer = printer
+
+	cmd := app.Command("offsets", "Describes offsets for a given topic and partition.")
+	listOffsetsCmd := cmd.Command("list", "List all offsets of a given topic and partition.").Action(c.listOffsets)
+	listOffsetsCmd.Flag("topic", "Kafka topic to dump").Required().StringVar(&c.topic)
+	listOffsetsCmd.Flag("type", "offset type to list").Default("end").EnumVar(&c.offsetType, "start", "committed", "end", "after_ms")
```

Comment:
nit to have the enum vals as `const`s. helps navigate the code faster (e.g. what's the place which does the `after_ms` processing)